packages/itwinui-css/src/input/input.scss

```diff
@@ -1,70 +1,76 @@
+  --_iui-input-border-color: var(--iui-color-foreground-4);
+  --_iui-input-text-color: var(--iui-color-foreground-2);
+
   @include iui-reset;
   @include iui-focus($offset: -2px, $thickness: 2px);
   width: 100%;
   font-family: inherit;
-  font-size: $iui-font-size;
-  font-weight: $iui-font-weight-normal;
-  line-height: $iui-line-height;
-  border-radius: $iui-border-radius;
+  font-size: var(--iui-font-size-1);
+  font-weight: var(--iui-font-weight-normal);
+  line-height: var(--iui-size-l);
+  border-radius: var(--iui-border-radius-1);
   appearance: none;
-  padding: $iui-component-padding-vertical $iui-component-padding-horizontal;
-  min-height: $iui-component-height;
-  color: var(--iui-text-color);
-  background-color: var(--iui-color-background-1);
-  border: 1px solid rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-4));
-  @media (prefers-reduced-motion: no-preference) {
-    transition: border-color $iui-speed-fast ease-out;
-  }
-
-  &.iui-small {
-    padding-top: $iui-component-padding-vertical-small;
-    padding-bottom: $iui-component-padding-vertical-small;
-    min-height: $iui-component-height-small;
-  }
-
-  &.iui-large {
-    padding-top: $iui-component-padding-vertical-large;
-    padding-bottom: $iui-component-padding-vertical-large;
-    min-height: $iui-component-height-large;
-    font-size: $iui-font-size-leading;
-  }
+  min-height: var(--_iui-input-min-height);
+  padding-block: var(--_iui-input-padding-block);
+  padding-inline: var(--iui-size-s);
+  color: var(--_iui-input-text-color);
+  background-color: var(--_iui-input-background-color);
+  border: 1px solid var(--_iui-input-border-color);
+  transition: border-color var(--iui-duration-1) ease-out;
+  box-sizing: border-box;
+  @include iui-input-size;
 
   &::placeholder {
+    --_iui-input-text-color: var(--iui-color-foreground-5);
     user-select: none;
-    color: var(--iui-text-color-placeholder);
+    color: var(--_iui-input-text-color);
   }
 
   &:autofill {
-    @include iui-input-autofill;
+    --_iui-input-background-color: var(--iui-color-background-subtle-primary);
+    --_iui-input-border-color: rgba(var(--iui-color-foreground-primary-rgb), var(--iui-opacity-1));
+    --_iui-input-text-color: rgba(var(--iui-color-foreground-primary-rgb), var(--iui-opacity-1));
   }
 
   &:hover {
-    border-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-2));
-    @media (prefers-reduced-motion: no-preference) {
-      transition: border-color $iui-speed-fast ease-out;
-    }
+    --_iui-input-border-color: var(--iui-color-foreground-2);
   }
 
   &[disabled] {
-    background-color: var(--iui-color-background-disabled);
-    border-color: var(--iui-color-background-disabled);
+    --_iui-input-background-color: var(--iui-color-background-disabled);
+    --_iui-input-border-color: var(--iui-color-background-disabled);
+    --_hover-color: var(--iui-color-background-disabled); // RENAME
```

Comment:
I don't know why it says rename, this variable has been around for 6 months

Reply:
> > > * `icon-color` -> `foreground-n`
> > > * `text-color` -> `foreground-n`
> 
> > Having aliases for these sounds nice, but it just gets complicated with the different variations of icons and text, and in some cases the variables start getting misused. Whereas with a single foreground scale, you only need to look at the documentation once and it just makes sense after that

Previously I knew, if I need to style icon, i use `icon-` variables, if I need to apply styling to text, i use `text-` variables. I dont think there were anymore variations of these groups. I think we even had a little different opacity for fill of icon and for text color at some point.

Can I see documentation, so it would make sense to me?

> 
> > > * `font-size-title` -> `font-size-4` (this one is pretty cruel, previously it was much better to guess what to use where we were reusing some of the common naming for different sizes)
> 
> > i don't agree with this in particular. previously the font-size variables seemed completely random to me, whereas now, `1` is the base font size, `0` is the smaller one, and `2`, `3`, etc are progressively getting larger.

So how I am supposed to know which of the number to use if I want to have a title of the section? Or which number is the `default` for general text in my app?
Shouldnt we at least start numbering from `1` so this is at least synced in some kind of a way with heading tags of html?
Does this mean our react components will become `FontElement2` too?

> 
> > > * even the baseline change, I do feel having a baseline variable would be so much better
> 
> > how does having a baseline variable help? we never followed the baseline grid anyway, and now it's just part of our normal spacing scale.

I know products have started to adopt baseline vs horizontal spacing recommendations after lots of years of pushing this, so they have their padding set to use `baseline` and `space` variables accordingly. So having an easier replace of the variables might look as an easier transition to the major release.
And in different variable it might be easier to explain, why they should use always `12px` instead of `16px` for vertical default spacing.

Reply:
Maybe rename it to meet our standards because it will never be renamed.

Reply:
Renamed